### DISCUSSION TITLE
求数据来源

### DISCUSSION THREAD

**Discussion by imingz** [ChangeSuger/fuxuan-archive#7] [Q&A] [answered]
想请问一下作者，这个 datas 的成就数据是您自己做的吗？还是其他地方获取的。
顺便一提，国际化好像只有数据部分，界面 UI 并没有国际化

**Comment by ChangeSuger** [comment, marked as answer]
成就数据是基于文本解包数据生成的，原始数据来源为 [StarRailData](https://github.com/Dimbreath/StarRailData) 

因为原始数据自带了所有语言的文本，所以数据部分有国际化；UI 的国际化需要额外实现，暂时还没开工 /(ㄒoㄒ)/~~

**Reply by lilliabi** [reply]
大佬你好，原始数据你是经过怎样的处理呢？

**Reply by ChangeSuger** [reply]
和成就相关的只有两个文件，数据结构也比较简单，基本没怎么处理原始数据，就根据需要提取了有用的部分，然后加了一些额外的信息。

使用的话就是根据 Hash 值查 textMap 获取对应语言的文本，然后用正则表达式对文本进行处理，文本处理可以参考 [decodeText](https://github.com/ChangeSuger/fuxuan-archive/blob/main/src/utils/decodeText.ts)。
